### PR TITLE
A benchmark to evaluate overhead of unschedulable pods

### PR DESCRIPTION
**Pull request by ahg-g**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
Adds a scheduler benchmark that evaluates the overhead of unschedulable pods.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/cc @Huang-Wei @alculquicondor

### CODE REVIEW

File: test/integration/scheduler_perf/scheduler_perf_test.go [hunk 1]
```diff
@@ -149,15 +150,15 @@ func perfScheduling(test testCase, b *testing.B) []DataItem {
...
-	total := 0
 	for _, p := range test.InitPods {
 		if err := createPods(setupNamespace, p, clientset); err != nil {
 			b.Fatal(err)
 		}
-		total += p.Num
-	}
-	if err := waitNumPodsScheduled(b, total, podInformer, setupNamespace); err != nil {
-		b.Fatal(err)
+		if !p.SkipWaitingUntilScheduled {
```
Comment (alculquicondor) [inline test/integration/scheduler_perf/scheduler_perf_test.go:157]: nit: keep this outside loop.

I feel like it could open other test cases involving uni directional affinity.

File: test/integration/scheduler_perf/config/pod-large-cpu.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  generateName: pod-
+spec:
+  containers:
+  - image: k8s.gcr.io/pause:3.2
+    name: pause
+    ports:
```
Comment (alculquicondor) [inline test/integration/scheduler_perf/config/pod-large-cpu.yaml:9]: why is this necessary?

File: test/integration/scheduler_perf/scheduler_perf_test.go [hunk 3]
```diff
@@ -90,6 +90,7 @@ type podCase struct {
 	PodTemplatePath                   *string
 	PersistentVolumeTemplatePath      *string
 	PersistentVolumeClaimTemplatePath *string
+	SkipWaitingUntilScheduled         bool
```
Comment (alculquicondor) [inline test/integration/scheduler_perf/scheduler_perf_test.go:93]: nit: SkipWaitUntilScheduled

**Review by alculquicondor — COMMENTED** [review]
just nits

File: test/integration/scheduler_perf/config/pod-large-cpu.yaml (hunk 2 above)
Comment (ahg-g) [reply test/integration/scheduler_perf/config/pod-large-cpu.yaml:9]: I just copied the default pod template.

File: test/integration/scheduler_perf/scheduler_perf_test.go (hunk 1 above)
Comment (ahg-g) [reply test/integration/scheduler_perf/scheduler_perf_test.go:157]: ok, how about we move the parameter outside the case then and have a single flag for all init pods, otherwise this is not really going to work well.

Comment (alculquicondor) [reply test/integration/scheduler_perf/scheduler_perf_test.go:157]: oh I missed that the parameter is inside the case. We an leave it like this for now until we have a reason to change it.

Comment (ahg-g) [reply test/integration/scheduler_perf/scheduler_perf_test.go:157]: I think we should do it outside, we can't enforce which exact pods we are waiting for in the loop unless each case creates the pods in a different namespace.

File: test/integration/scheduler_perf/scheduler_perf_test.go (hunk 3 above)
Comment (ahg-g) [reply test/integration/scheduler_perf/scheduler_perf_test.go]: done.

File: test/integration/scheduler_perf/scheduler_perf_test.go (hunk 1 above)
Comment (alculquicondor) [reply test/integration/scheduler_perf/scheduler_perf_test.go]: sg

Comment (ahg-g) [reply test/integration/scheduler_perf/scheduler_perf_test.go]: squashed

**Review by alculquicondor — COMMENTED** [review]
/lgtm
